std/functional.d

```diff
@@ -260,165 +208,342 @@ unittest
+        == "alias foo = args[0];\nalias b = args[1];\n");
+    static assert(strLambdaAliases!(2, "foo", "ping")
+        == "alias foo = args[0];\nalias ping = args[1];\n");
+
+    // reallocation
+    static assert(strLambdaAliases!(1, "a123456789b123456789")
+        == "alias a123456789b123456789 = args[0];\n");
+    static assert(strLambdaAliases!(1, "a123456789b123456789c123456789d123456789")
+        == "alias a123456789b123456789c123456789d123456789 = args[0];\n");
+
+    // large argc
+    static assert(strLambdaAliases!(63) == strLambdaAliases!(35));
 }
 
+///
 unittest
 {
-    static assert(!_ctfeMatchUnary("sqrt(ё)", "ё"));
-    static assert(!_ctfeMatchUnary("ё.sqrt", "ё"));
-    static assert(!_ctfeMatchUnary(".ё+ё", "ё"));
-    static assert(!_ctfeMatchUnary("_ё+ё", "ё"));
-    static assert(!_ctfeMatchUnary("ёё", "ё"));
-    static assert(_ctfeMatchUnary("a+a", "a"));
-    static assert(_ctfeMatchUnary("a + 10", "a"));
-    static assert(_ctfeMatchUnary("4 == a", "a"));
-    static assert(_ctfeMatchUnary("2==a", "a"));
-    static assert(_ctfeMatchUnary("1 != a", "a"));
-    static assert(_ctfeMatchUnary("a!=4", "a"));
-    static assert(_ctfeMatchUnary("a< 1", "a"));
-    static assert(_ctfeMatchUnary("434 < a", "a"));
-    static assert(_ctfeMatchUnary("132 > a", "a"));
-    static assert(_ctfeMatchUnary("123 >a", "a"));
-    static assert(_ctfeMatchUnary("a>82", "a"));
-    static assert(_ctfeMatchUnary("ё>82", "ё"));
-    static assert(_ctfeMatchUnary("ё[ё(ё)]", "ё"));
-    static assert(_ctfeMatchUnary("ё[21]", "ё"));
+    // generated param names
+    alias sum = stringLambda!("a + b + c", 3);
+    assert(sum(1, 2, 3) == 6);
+
+    // explicit param names
+    alias prod = stringLambda!("x * y * z * w", "x", "y", "z", "w");
+    assert(prod(2,3,4,5) == 120);
+
+    // library function invocation
+    alias least = stringLambda!("min(a,b,c)", 3);
+    assert(least(5, 3, 7) == 3);
+}
+
+// stringLambda edge cases
+unittest
+{
+    import std.typecons : staticIota;
+
+    // use all 26 generated params
+    alias many = stringLambda!("[a,b,y,z]", 26);
+    assert(many(staticIota!(0, 26)) == [0,1,24,25]);
+
+    // use all 26 generated params plus a 27'th 'unnamed' param
+    alias more = stringLambda!("[a,b,y,z,args[26]]", 27);
+    assert(more(staticIota!(0, 27)) == [0,1,24,25,26]);
+
+    // use a mix of explicit params and args[n]
+    alias mix = stringLambda!("[x,y,args[2],args[3]]", 4, "x", "y");
+    assert(mix(1,2,3,4) == [1,2,3,4]);
+
+    // use a mix of explicit and implicit params
+    alias mix2 = stringLambda!("[x,y,c,d]", 4, "x", "y");
+    assert(mix(1,2,3,4) == [1,2,3,4]);
 }
 
-// returns 1 if $(D fun) is trivial binary function
-private uint _ctfeMatchBinary(string fun, string name1, string name2)
+// returns true if `fun` contains symbols other than its parameters
+private bool strLambdaNeedsImports(string fun, const(string[]) names...)
 {
     if (!__ctfe) assert(false);
```

Comment:
@greenify I guess I could, but I don't think that's really necessary - this is a private symbol which is only used in one (non-`unittest`) place.

The `if(!_ctfe) assert(false);` idiom is used many other places in Phobos, and I don't recall every seeing a message included.